### PR TITLE
Add a `-timings` command-line flag

### PR DESCRIPTION
**Pull request by vbgl**

### CODE REVIEW

**Review by bgregoir — COMMENTED** [review]
I think it is a first good step. I have the feeling that it would be more informative to have the time taken by the transformation. I hope this is possible to do. It should be doable by storing the previous printed time.

**Comment by vbgl** [comment]
I’ll add a print of the elapsed time.

**Comment by vbgl** [comment]
I’ve just rebased and added display of elapsed time.

File: compiler/src/utils.ml [hunk 1]
```diff
@@ -394,6 +394,45 @@ let hierror ~loc ?funname ~kind ?sub_kind ?(internal=false) =
...
+let pp_now =
+  let open Unix in
+  let timestamp = ref (-1.) in
+  let pp_elapsed fmt now =
+    let old = !timestamp in
+    if old >= 0. then begin
+      let diff = now -. old in
+      let h, m, s, c = hmsc diff in
+      Format.fprintf fmt "|";
+      let g = false in
```
Comment (bgregoir) [inline compiler/src/utils.ml:422]: This two line are strange, can we replace them by let g = h > 0

File: compiler/src/utils.ml [hunk 2]
```diff
@@ -394,6 +394,45 @@ let hierror ~loc ?funname ~kind ?sub_kind ?(internal=false) =
       raise (HiError err))
 
 
+(* -------------------------------------------------------------------- *)
+(** Splits a time in seconds into hours, minutes, seconds, and centiseconds.
+  Number of hours must be below one hundred. *)
+let hmsc f =
```
Comment (bgregoir) [inline compiler/src/utils.ml:400]: This look like localtime, the diff is in the c?
Maybe also in no day but only our.

**Review by bgregoir — COMMENTED** [review]
Small extra comment, but I think all we discussed before is here.

File: compiler/src/utils.ml (hunk 1 above)
Comment (vbgl) [reply compiler/src/utils.ml:422]: sure.

File: compiler/src/utils.ml (hunk 2 above)
Comment (vbgl) [reply compiler/src/utils.ml:400]: There is some timezone for with a localtime of 1s is 11:00pm. This is not what I want here…

File: compiler/src/utils.ml (hunk 1 above)
Comment (vbgl) [reply compiler/src/utils.ml]: Done.

**Review by bgregoir — APPROVED** [review]

**Comment by bgregoir** [comment]
Thank Vincent.

**Pull request merged**